rfcs/0010-eaglesong/eaglesong.py

```diff
@@ -0,0 +1,183 @@
+    0xee37f6bc ,  0x27116f48 ,  0x5e9ec45a ,  0x8ea2a51f ,  0xa5573db7 ,  0xa746d036 ,  0x486b4768 ,  0x5b438f3b ,
+    0x18c54a5c ,  0x64fcf08e ,  0xe993cdc1 ,  0x35c1ead3 ,  0x9de07de7 ,  0x321b841c ,  0x87423c5e ,  0x071aa0f6 ,
+    0x962eb75b ,  0xbb06bdd2 ,  0xdcdb5363 ,  0x389752f2 ,  0x83d9cc88 ,  0xd014adc6 ,  0xc71121bb ,  0x2372f938 ,
+    0xcaff2650 ,  0x62be8951 ,  0x56dccaff ,  0xac4084c0 ,  0x09712e95 ,  0x1d3c288f ,  0x1b085744 ,  0xe1d3cfef ,
+    0x5c9a812e ,  0x6611fd59 ,  0x85e46044 ,  0x1981d885 ,  0x5a4c903f ,  0x43f30d4b ,  0x7d1d601b ,  0xdd3c3391 ,
+    0x030ec65e ,  0xc12878cd ,  0x72e795fe ,  0xd0c76abd ,  0x1ec085db ,  0x7cbb61fa ,  0x93e8dd1e ,  0x8582eb06 ,
+    0x73563144 ,  0x049d4e7e ,  0x5fd5aefe ,  0x7b842a00 ,  0x75ced665 ,  0xbb32d458 ,  0x4e83bba7 ,  0x8f15151f ,
+    0x7795a125 ,  0xf0842455 ,  0x499af99d ,  0x565cc7fa ,  0xa3b1278d ,  0x3f27ce74 ,  0x96ca058e ,  0x8a497443 ,
+    0xa6fb8cae ,  0xc115aa21 ,  0x17504923 ,  0xe4932402 ,  0xaea886c2 ,  0x8eb79af5 ,  0xebd5ea6b ,  0xc7980d3b ,
+    0x71369315 ,  0x796e6a66 ,  0x3a7ec708 ,  0xb05175c8 ,  0xe02b74e7 ,  0xeb377ad3 ,  0x6c8c1f54 ,  0xb980c374 ,
+    0x59aee281 ,  0x449cb799 ,  0xe01f5605 ,  0xed0e085e ,  0xc9a1a3b4 ,  0xaac481b1 ,  0xc935c39c ,  0xb7d8ce7f ]
+
+    # bit matrix
+    new = [0 for i in range(0,16)]
+    for j in range(0, 16):
+        for k in range(0, 16):
+            new[j] = new[j] ^ (state[k] * bitmatrix[k][j])
+        new[j] = new[j] & 0xffffffff # truncate to 32 bits, if necessary
+    state = new
+
+    # circulant multiplication
+    for i in range(0, 16):
+        acc = 0
+        for j in range(0, 3):
+            acc = acc ^ (state[i] << coefficients[i][j]) ^ (state[i] >> (32-coefficients[i][j]))
+        state[i] = acc & 0xffffffff # truncate to 32 bits, if necessary
+
+    # constants injection
+    for i in range(0, 16):
+        state[i] = state[i] ^ injection_constants[index*16 + i]
+
+    # add / rotate / add
+    for i in range(0, 8):
+        state[2*i] = (state[2*i] + state[2*i+1]) & 0xffffffff # truncate to 32 bits, if necessary
+        state[2*i] = (state[2*i] >> 24) ^ ((state[2*i] << 8) & 0xffffffff) # shift bytes
+        state[2*i+1] = (state[2*i+1] >> 8) ^ ((state[2*i+1] << 24) & 0xffffffff) # shift bytes
+        state[2*i+1] = (state[2*i] + state[2*i+1]) & 0xffffffff # truncate to 32 bits, if necessary
+
+    return state
+
+def EaglesongSponge( input_bytes, num_output_bytes, delimiter ):
+    # parameters
+    capacity = 256 # must be multiple of 32
+    rate = 256 # must be multiple of 32
+
+    state = [0 for i in range(0, 16)]
+
+    # absorbing
+    for i in range(0, ((len(input_bytes)+1)*8+rate-1) // rate):
+        for j in range(0, rate//32):
+            integer = 0
+            for k in range(0, 4):
+                if i*rate//8 + j*4 + k < len(input_bytes):
+                    integer = (integer << 8) ^ input_bytes[j*4 + k]
```

Comment:
bug? using the wrong index:

```suggestion
                    integer = (integer << 8) ^ input_bytes[i*rate//8 + j*4 + k]
```

Reply:
You're right, great catch! I guess the utility of having two independent implementations is undermined when you base one off the other ...